zulipterminal/model.py

```diff
@@ -380,13 +383,12 @@ def update_stream_message(self, topic: str, message_id: int,
         display_error_if_present(response, self.controller)
         return response['result'] == 'success'
 
-    def fetch_custom_emojis(self) -> NamedEmojiData:
-        response = self.client.get_realm_emoji()
+    def process_realm_emoji_data(self) -> NamedEmojiData:
         custom_emojis = {emoji['name']: {'code': emoji_code,
                                          'type': 'realm_emoji'}
-                         for emoji_code, emoji in response['emoji'].items()
+                         for emoji_code, emoji in
+                         self.initial_data['realm_emoji'].items()
                          if not emoji['deactivated']}  # type: NamedEmojiData
-        display_error_if_present(response, self.controller)
         return custom_emojis
```

Comment:
This is difficult to test and reuse since you're not allowing the function to vary based on some other data - ie. passing in a parameter, like initial_data emoji vs event emoji.